src/browser/BLECentralPlugin.js

```diff
@@ -1,63 +1,194 @@
+   
 function notSupported() {
     console.log('BLE is not supported on the browser');
 }
 
+
+function getDeviceId(device_id) {
+    return device_id.startsWith('0x') ? parseInt(device_id) : device_id;
+}
+
+
 module.exports = {
+
+    devices: new Map(),
+    scanning: null,
+
     scan: function(services, seconds, success, failure) {
-        notSupported();
-        if (failure) failure();
-    },
-    startScan: function(services, success, failure) {
-        notSupported();
-        if (failure) failure();
+
+        if (!navigator.bluetooth) {
+            failure('Bluetooth is not supported on this browser.');
+            return;
+        }
+
+        this.scanning = true;
+
+        let options = {};
+
+        if (services && services.length) {
+            options.filters = [{
+                services: services.map(getDeviceId)
+            }];
+        } else {
+            options.acceptAllDevices = true;
+        }
+
+        console.log('requestDevice', options);
+        navigator.bluetooth.requestDevice(options).then(device => {
+            if (this.scanning) {  
+                return device.gatt.connect().then(server => {
+                    this.devices.set(device.id, {
+                        device: device,
+                        server: server
+                    })
+                    success(device);
```

Comment:
I'd suggest calling this immediately when `requestDevice` succeeds and leaving connect to the `connect` method later.

Also, it'd be great to stay aligned with the existing types returned here as much as possible. E.g., this should probably return an `{ id: device.id }` and nothing else.